.buildkite/publish/publish-common.sh

```diff
@@ -21,7 +23,24 @@ if [[ "${USE_SNAPSHOT:-}" == "true" ]]; then
   export VERSION="${VERSION}-SNAPSHOT"
 fi
 
+if [[ "${MANUAL_RELEASE:-}" == "true" ]]; then
+  export ORIG_VERSION=$(buildkite-agent meta-data get orig_version)
+  IFS='.' read -ra VERSION_PARTS <<< "$ORIG_VERSION"
+  PATCH_PART=${VERSION_PARTS[2]}
+  LAST_PATCH="$((PATCH_PART - 1))"
+  LAST_VERSION="${VERSION_PARTS[0]}.${VERSION_PARTS[1]}.${LAST_PATCH}"
```

Comment:
> What if we need to manually publish 8.16.3, but the VERSION file has been already bumped to 8.17.0? 

This shouldn't ever happen. As soon as we Feature Freeze for an upcoming `.0` release, that minor gets its own branch. So as we prep to release 8.16.0 next month (FF mid Oct), we'll create an `8.16` branch. And we wouldn't ever be bumping the `8.16` branch's `VERSION` file to `8.17.0` - that will happen _usually_ on `main`, but for `8.17.0` specifically it'll happen on `8.x` (because of the prep for 9.0.0 major bump).

Also, this manual release job will be ONLY for out-of-band releases. The Unified Release build will handle `8.16.3`. This would be if we were preparing for an 8.16.3 Elastic-wide, but had a sudden, urgent need to release a bugfix a week before the planned release date. So if our `VERSION` file was already at `8.16.3`, we'd calculate `LAST_VERSION` as `8.16.2`, and then append the `build<timestamp>` suffix.

I'll take a note to add some more comments in here, so that future reviewers can at least look back to this PR discussion.